pkg/jobs/registry.go

```diff
@@ -1282,6 +1286,33 @@ func (r *Registry) adoptionDisabled(ctx context.Context) bool {
 	return false
 }
 
+// A single job should not toggle its idleness more than twice per-minute as it
+// is logged and may write to persisted job state in the future
+func (r *Registry) MarkIdle(job *Job, isIdle bool) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	if aj, ok := r.mu.adoptedJobs[job.ID()]; ok {
+		payload := job.Payload()
+		jobType := payload.Type()
+		jm := r.metrics.JobMetrics[jobType]
+		if aj.isIdle != isIdle {
+			log.Infof(r.serverCtx, "%s job %d: toggling idleness to %+v", jobType, job.ID(), isIdle)
```

Comment:
I would probably doe if log.V(1) -- this just seems to be too verbose. We don't seem to log normal job behaviors, transitions in states, etc.  This seems to be normal.

Reply:
So I think I argued for logging these state transitions in the issue. We log, today, when a job starts running ("stepping though state running..."). This is useful since you can now look at a log and go "aha, the job started running at 123, and as 124 we see these things explode". If a job is moving from idle/parked into active work, that is potentially equally notable to when it initially started running, if idleness is defined on a long enough time-scale.